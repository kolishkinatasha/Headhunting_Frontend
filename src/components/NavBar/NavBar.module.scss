.box {
  display: flex;
  justify-content: space-between;
  padding-bottom: 10px;
  padding-left: 10px;
  padding-right: 10px;
  border-bottom: 0.1px solid #fff;
  margin-top: -30px;
  box-shadow: 0 5px 5px -7px rgba(0, 87, 255, .5);
}
.buttonBox {
  display: flex;
  flex-direction: row;
  width: fit-content;
  align-items: center;
}
.title {
  margin: 20px 0px;
}
.buttons {
  margin: 10px 0px 10px 10px;
}
