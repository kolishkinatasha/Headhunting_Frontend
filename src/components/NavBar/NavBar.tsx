import React from 'react';
import { Button, Typography } from '@ff/ui-kit';
import ApplicationCard from 'components/Modals/ApplicationCard';
import { observer } from 'mobx-react';
import { NavLink } from 'react-router-dom';
import ModalWrapper from 'components/layouts/Modal/ModalWrapper';

import classes from './NavBar.module.scss';
import profile from '../../stores/profile';

const allowRoles = ['Администратор', 'Главный рекрутер'];

const button = (
  <Button variant="outline" type="primary" className={classes.buttons} style={{ width: '185px' }}>
    Подать заявку
  </Button>
);

const NavBar: React.FC = observer(() => (
  <div className={classes.box}>
    <Typography.Title level={2} node="div" className={classes.title}>
      Заявки на подбор персонала
    </Typography.Title>
    <div className={classes.buttonBox}>
      <ModalWrapper button={button}>
        <ApplicationCard />
      </ModalWrapper>
      { allowRoles.includes(profile.role)
      && (
      <NavLink to="/services/work/applications/register">
        <Button variant="fill" type="primary" className={classes.buttons}>
          Реестор всех заявок
        </Button>
      </NavLink>
      )}
    </div>
  </div>
));

export default NavBar;
