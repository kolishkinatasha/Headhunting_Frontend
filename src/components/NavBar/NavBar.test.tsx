import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';

import NavBar from './NavBar';

describe('NavBar', () => {
  it('should be the same', () => {
    const history = createMemoryHistory();
    const app = render(<Router history={history}><NavBar /></Router>);
    expect(app).toMatchSnapshot();
  });
});
