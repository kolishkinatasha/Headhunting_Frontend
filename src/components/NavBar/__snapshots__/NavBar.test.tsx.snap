// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should be the same 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="box"
      >
        <h2
          class="fns-typography fns-typography_h2 title"
        >
          Заявки на подбор персонала
        </h2>
        <div
          class="buttonBox"
        >
          <div
            class="modalBox"
          >
            <div
              role="button"
            >
              <button
                class="fns-button fns-button_outline fns-button_primary buttons"
                style="width: 185px;"
                type="button"
              >
                <span>
                  Подать заявку
                </span>
              </button>
            </div>
          </div>
          <a
            href="/services/work/applications/register"
          >
            <button
              class="fns-button fns-button_fill fns-button_primary buttons"
              type="button"
            >
              <span>
                Реестор всех заявок
              </span>
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      style="position: absolute; top: 0px; left: 0px; width: 100%;"
    />
  </body>,
  "container": <div>
    <div
      class="box"
    >
      <h2
        class="fns-typography fns-typography_h2 title"
      >
        Заявки на подбор персонала
      </h2>
      <div
        class="buttonBox"
      >
        <div
          class="modalBox"
        >
          <div
            role="button"
          >
            <button
              class="fns-button fns-button_outline fns-button_primary buttons"
              style="width: 185px;"
              type="button"
            >
              <span>
                Подать заявку
              </span>
            </button>
          </div>
        </div>
        <a
          href="/services/work/applications/register"
        >
          <button
            class="fns-button fns-button_fill fns-button_primary buttons"
            type="button"
          >
            <span>
              Реестор всех заявок
            </span>
          </button>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
